rules/DeadCode/PhpDoc/DeadReturnTagValueNodeAnalyzer.php

```diff
@@ -46,15 +49,15 @@ public function isDead(ReturnTagValueNode $returnTagValueNode, ClassMethod $clas
             $returnTagValueNode->type,
             $classMethod,
         )) {
-            return false;
+            return $returnTagValueNode->type instanceof IdentifierTypeNode && (string) $returnTagValueNode->type === 'void';
```

Comment:
@staabm this is the solution I can found right now, due to sub type and generic usage, union complexity :), `void` mostly a different part so should be ok for this case ;)